enclave_mgmt/compile_models.py

```diff
@@ -714,10 +857,46 @@ def collect_content_dfs(bucket, key):
             }
 
 
-def compile_models(data_bucket, data_key):
+def collect_event_data_dfs(events_bucket, events_key):
+
+    key_content_loads = events_key + "/content_loaded_v1.json"
+    key_ib_problem_attempts = events_key + "/ib_pset_problem_attempted_v1.json"
+    key_ib_input_submissions = events_key + "/ib_input_submitted_v1.json"
+
+    s3_client = boto3.client("s3")
+
+    content_loads_stream = s3_client.get_object(
+        Bucket=events_bucket,
+        Key=key_content_loads)
+    ib_problem_attempts_stream = s3_client.get_object(
+        Bucket=events_bucket,
+        Key=key_ib_problem_attempts)
+    ib_input_submissions_stream = s3_client.get_object(
+        Bucket=events_bucket,
+        Key=key_ib_input_submissions)
+
+    content_loads_data = pd.DataFrame(
+        json.loads(content_loads_stream["Body"].read())['data']
+    )
+    ib_problem_attempts_data = pd.DataFrame(
+        json.loads(ib_problem_attempts_stream["Body"].read())['data']
+    )
```

Comment:
I missed raising this when we paried as I forgot all about the downstream implications of deserializing Avro data where the schema has a Union of types (see separate comment in the fixtures). Depending on the approach we decide to take, if we push handling that down to the edge here this may need to be something like:

```python
    # Normalize union type for pset attempt response
    ib_problem_attempts_json = json.loads(
        ib_problem_attempts_stream["Body"].read()
    )['data']
    ib_problem_attempts_normalized = []
    for item in ib_problem_attempts_json:
        item["response"] = \
            item["response"]["string"] or item["response"]["array"]
        ib_problem_attempts_normalized.append(item)
    ib_problem_attempts_data = pd.DataFrame(
        ib_problem_attempts_normalized
    )
```